tiledb/sm/filesystem/ls_scanner.h

```diff
@@ -96,12 +80,6 @@ using FileFilter = std::function<bool(const std::string_view&, uint64_t)>;
   return true;
 }
 
-using DirectoryFilter = std::function<bool(const std::string_view&)>;
```

Comment:
After more thought, I decided to remove `DirectoryFilter`; it is used only in tests, and cannot be implemented efficiently on object storage backends.

Would appreciate another review.

Reply:
@shaunrd0 @teo-tsirpanis If this is causing any delays to [CORE-248](https://linear.app/tiledb/issue/CORE-248/ls-recursive-does-not-list-directories-in-object-storage) we should not do it right now. That task has the higher priority, we can always come back and refactor things afterwards.